src/phases/mapping.rs

```diff
@@ -0,0 +1,445 @@
+            explanation_line2: gui
+                .textbuf
+                .make_static_text(c"match unknown Doodle authors to your friends."),
+            explanation_line3: gui
+                .textbuf
+                .make_static_text(c"You can change the suggested mapping"),
+            explanation_line4: gui
+                .textbuf
+                .make_static_text(c"before you begin the migration."),
+            explanation_line5: gui
+                .textbuf
+                .make_static_text(c"Highlight a mapping with \u{E07D} "),
+            explanation_line6: gui
+                .textbuf
+                .make_static_text(c"and press \u{E000} to change it."),
+            press_a_continue: gui.textbuf.make_static_text(c"Press \u{E000} to continue"),
+            press_b_back: gui.textbuf.make_static_text(c"\u{E001} Back"),
+            press_x_clear: gui.textbuf.make_static_text(c"\u{E002} Clear"),
+            press_y_done: gui.textbuf.make_static_text(c"\u{E003} Finish"),
+            remapping: gui.textbuf.make_static_text(c"Remapping"),
+            section_doodles: gui.textbuf.make_static_text(c"Doodle pals"),
+            section_friends: gui.textbuf.make_static_text(c"3DS Friends"),
+            dont_map: gui.textbuf.make_static_text(c"<don't map>"),
+            apt,
+            gfx,
+            hid,
+            gui,
+        }
+    }
+
+    pub fn run(mut self, read: ReadResult) -> MigrationFlow<OldToNewPIDMapping> {
+        let mapping = RefCell::new(OldToNewPIDMapping::new());
+
+        Self::auto_match_by_mac(&mapping, &read);
+
+        for (pid, mii) in read.doodles.iter() {
+            self.names.insert(
+                *pid,
+                self.gui
+                    .textbuf
+                    .make_static_text(&CString::from_str(&mii.mii_name).unwrap_or_default()),
+            );
+        }
+        for (pid, mii) in read.friends.iter() {
+            self.names.insert(
+                *pid,
+                self.gui
+                    .textbuf
+                    .make_static_text(&CString::from_str(&mii.mii_name).unwrap_or_default()),
+            );
+        }
+
+        loop {
+            process(self.apt, self.gfx, self.hid)?;
+            self.begin_paint();
+            self.dialog_explanation();
+            self.end_paint();
+
+            if self.hid.keys_down().contains(KeyPad::A) {
+                break;
+            }
+        }
+
+        self.mapping_editor(&mapping, &read)?;
+
+        MigrationFlow::Continue(mapping.into_inner())
+    }
+
+    fn auto_match_by_mac(mapping: &RefCell<OldToNewPIDMapping>, read: &ReadResult) {
+        let mut mapping = mapping.borrow_mut();
+        for doodler in &read.doodles {
+            let mac = doodler.1.creator_mac_address;
+            for friend in &read.friends {
+                if friend.1.creator_mac_address == mac {
+                    mapping.insert(*doodler.0, *friend.0);
```

Comment:
```suggestion
        for (&doodler_pid, doodler_mii) in &read.doodles {
            for (&friend_pid, friend_mii) in &read.friends {
                if friend_mii.creator_mac_address == doodler_mii.creator_mac_address {
                    mapping.insert(doodler_pid, friend_pid);
```
Giving these values names and copying the PIDs is maybe more readable